js/views/conversation_view.js

```diff
@@ -577,33 +580,77 @@
       //   events up to its parent elements in the DOM.
       this.closeMenu();
 
-      const media = await Signal.Backbone.Conversation.fetchVisualMediaAttachments({
-        conversationId: this.model.get('id'),
-        WhisperMessageCollection: Whisper.MessageCollection,
+      // We fetch more documents than media as they don’t require to be loaded
+      // into memory right away. Revisit this once we have infinite scrolling:
+      const DEFAULT_MEDIA_FETCH_COUNT = 50;
+      const DEFAULT_DOCUMENTS_FETCH_COUNT = 150;
+
+      const conversationId = this.model.get('id');
+      const WhisperMessageCollection = Whisper.MessageCollection;
+      const rawMedia = await Signal.Backbone.Conversation.fetchVisualMediaAttachments({
+        conversationId,
+        count: DEFAULT_MEDIA_FETCH_COUNT,
+        WhisperMessageCollection,
+      });
+      const documents = await Signal.Backbone.Conversation.fetchFileAttachments({
+        conversationId,
+        count: DEFAULT_DOCUMENTS_FETCH_COUNT,
+        WhisperMessageCollection,
       });
-      const loadMessages = Signal.Components.PropTypes.Message
+
+      // NOTE: Could we show grid previews from disk as well?
```

Comment:
Is it the full image that we're showing, just scaled down, in each of the grid elements? Would definitely be good to start getting rigorous with thumbnail management.

Reply:
Yes, that’s correct, we are showing the full-size image scaled down due to lack of thumbnails.